### DISCUSSION TITLE
'Is a' Relationship in Conceptual Model

### DISCUSSION THREAD

**Discussion by mehdihadeli** [kgrzybek/modular-monolith-with-ddd#165] [Q&A] [answered]
Hi,
I have a question about your Conceptual model,
Why in this model `Member` has an Is a relationship to `User`?
Our `Member` class not inherits from `User` directly and my question is why you use this 'is a' relationship here

**Comment by kgrzybek** [comment, marked as answer]
Hi @mehdihadeli 

In Conceptual Model you model your concepts on higher level of abstraction without thinking about Object-Oriented Design / Functional Programming / other programming paradigm. You don't think about software, you **model your business concepts**. 

So _"is a"_ term it **is not** _"inheritance"_ from the OOD paradigm. It is just a description of relation in non-programming language. The reason is that kind of model should be understandable for non-technical people as well. 

In lower level (code) it could be modeled as you described (using inheritance) but it doesn't have to. Moreover, inheritance in general should be avoided (https://en.wikipedia.org/wiki/Composition_over_inheritance) because inheritance introduces the highest level of coupling between classes.

**Reply by mehdihadeli** [reply]
@kgrzybek Thanks for your guide.

**Reply by kgrzybek** [reply]
No problem, just "mark as answer" if it was helpful, please :)
